packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -1175,21 +1189,24 @@ export class AsgCapacityProvider extends Construct {
 class MaybeCreateCapacityProviderAssociations implements IAspect {
   private scope: Construct;
   private id: string;
-  private capacityProviders: string[]
-  private resource?: CfnClusterCapacityProviderAssociations
+  private capacityProviders: string[];
+  private resource?: CfnClusterCapacityProviderAssociations;
+  private defaultCapacityProviderStrategy?: CapacityProviderStrategy[];
+
 
-  constructor(scope: Construct, id: string, capacityProviders: string[] ) {
+  constructor(scope: Construct, id: string, capacityProviders: string[], defaultCapacityProviderStrategy?: CapacityProviderStrategy[]) {
     this.scope = scope;
     this.id = id;
     this.capacityProviders = capacityProviders;
+    this.defaultCapacityProviderStrategy = defaultCapacityProviderStrategy;
   }
 
   public visit(node: IConstruct): void {
     if (node instanceof Cluster) {
       if (this.capacityProviders.length > 0 && !this.resource) {
         const resource = new CfnClusterCapacityProviderAssociations(this.scope, this.id, {
           cluster: node.clusterName,
-          defaultCapacityProviderStrategy: [],
+          defaultCapacityProviderStrategy: this.defaultCapacityProviderStrategy || [],
           capacityProviders: Lazy.list({ produce: () => this.capacityProviders }),
```

Comment:
should we validate that the list of capacity providers includes all the capacity providers in the defaultCapacityProviderStrategy? What happens if a customer tries to include a capacity provider in the latter list without including it in the former?

Reply:
We should make sure to test out these scenarios and if any of them fail, add validations and corresponding unit tests.